### PR TITLE
Remove HoustonConfig implemenation in favor of environment variables

### PR DESCRIPTION
**Pull request by mmulich**
## Pull Request Overview

It's important to distinguish between Houston Config and Site Settings. General Configuration, which the Houston Config and Flask Config are a part of, are application settings that need to be configured before app startup, configuration that is read-only and shouldn't be changed by the frontend users (e.g. database connection string), and require restart to take effect. Site Settings on the other hand are settings the site admins could change and expect the site to utilize without restart (e.g. site name, email address). So two very different things that most notably has to do with end-user mutability and service restart.

AFAICT we are not using Houston Config. I say this with some degree of certainty, because I know I'm not using it in the deployment code. And last I recall, it's not being used in the docker-compose setup.

The primary means of configuring the application is through **environment variables**. In development we do this through docker-compose, which happens to use the `.env` file for ease of use. In deployment on azure the settings come out of `etcd`, a key store database... Think of it like the `.env` file equivalent, but not on the filesystem. The use of environment variables is what gives us the ability to **deploy the application code for different instances without changing the code**.

I heard the use case for Houston Config is to allow for the deployment of multiple sites using the same codebase. I'd say that is already addressed and does not use Houston Config to achieve that end. The confusion seems to be around the flat `.env` configuration file as something that is needed for every deployment. This is not the case. The `.env` file is for ease of use in defining environment variables. The `.env` provided in the repo is for developer use. (note, it is ignored by docker; see `.dockerignore`).

We've got this HoustonConfig implementation that requires the database before application instantiation and startup, but it does nothing except occasionally stand in the way. This change removes it because we don't use it and the use case it was meant to solve is addressed in a different and probably more robust way.

### Model Updates

- `HoustonConfig`: removed

### Database Updates

The `houston_config` table has been removed.

### REST API Updates

The `/api/v1/config/houston` endpoints have been removed. These AFAICT are not used by frontend code.

### CODE REVIEW

File: migrations/versions/9a6d8b560a5a_.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    Upgrade Semantic Description:
+        ENTER DESCRIPTION HERE
+    """
+    # ### commands auto generated by Alembic - please adjust! ###
+    with op.batch_alter_table('houston_config', schema=None) as batch_op:
+        batch_op.drop_index('ix_houston_config_created')
+        batch_op.drop_index('ix_houston_config_indexed')
+        batch_op.drop_index('ix_houston_config_updated')
+
+    op.drop_table('houston_config')
```
Comment (karenc) [inline migrations/versions/9a6d8b560a5a_.py:31]: Wondering if we should print out all the data in case the table isn't empty?

**Review by karenc — APPROVED** [review]
The changes look good to me, but I don't know whether it's ok to remove HoustonConfig or not.  Michael said the houston config table is empty in the zebra migration database so may be it's not used?

**Comment by hwindsor** [comment]
I don't have a strong opinion on if this is right or wrong. 
Generally less code is good but I'm going to defer to @bluemellophone on this as (I think) he implemented this.

File: migrations/versions/9a6d8b560a5a_.py (hunk 1 above)
Comment (mmulich) [reply migrations/versions/9a6d8b560a5a_.py:31]: I suppose it wouldn't hurt. 😸

**Comment by mmulich** [comment]
The use-case brought up in today's discussion was around being able to configure permissions, which would be a complex data-type. This type of information wouldn't necessarily be something defined in a string value that one would put in an environment variable.

---

My proposal would be to create a DSL (Domain Specific Language) for the permissions rules. The permissions are then written into a file in this DSL. The file is pointed at from and environment variable and the system loads it on startup.

Alternatively, but in the same vain, instead of the DSL we continue to write it as Python that gets imported from some arbitrary location. I'm not a huge fan of using Python as a configuration language, because it can get you into strange situations that become difficult to maintain (e.g. yaml, ini, toml are good languages, because they don't let you do conditional statements; thus wysiwyg).

**Review by bluemellophone — APPROVED** [review]
LGTM

**Comment by bluemellophone** [comment]
The entire purpose, as far as I implemented it, was to allow for Flask-specific and application-specific configuration to be stored in the database.  I believe this was originally added to support the GitLab integration, where you could specify the GitLab variables once, save them to the database, and then they would be retrieved every time the system loaded.  The benefit was that the configuration values would "travel with" the application.  The problem is 1) this design was last used over 6 months ago and 2) we don't really use an SQLite-forward design anymore for the database.  I can't think of anything in Houston that removing this feature would actively break something.  I agree with @mmulich that permission rules should be defined separately.

I'm also not entirely sure why HoustonConfigs have to be removed (other than it makes the database migrations more complicated).  I'm personally on the side of keeping them in because it is a working feature (app configs stored in DB), but I also don't see a reason to keep them around either.

**Comment by colinwkingen** [comment]
@mmulich I'm ready to merge this, but there is a failure happening:
```

ERROR: for celery_beat  Container "70e0b58aa7ad" is unhealthy.
Encountered errors while bringing up the project.
Error: Process completed with exit code 1.
```

**Comment by karenc** [comment]
@colinwkingen It's fixed now

**Pull request merged**